src/vellum/workflows/events/tests/test_event.py

```diff
@@ -419,3 +420,30 @@ class MockWorkflow(BaseWorkflow[MockInputs, BaseState]):
 )
 def test_event_serialization(event, expected_json):
     assert not DeepDiff(event.model_dump(mode="json"), expected_json)
+
+
+def test_parent_context__deserialize_from_json__invalid_parent_context():
+    # GIVEN an event with a parent context that Vellum is introducing in the future
+    data = {
+        "foo": "bar",
+        "parent": {
+            "type": "SOME_FUTURE_ENTITY",
+            "span_id": "123e4567-e89b-12d3-a456-426614174000",
+            "some_randome_field": "some_random_value",
+            "parent": None,
+        },
+    }
+
+    # AND a dataclass that references the parent context
+    class MyData(UniversalBaseModel):
```

Comment:
Doing a custom dataclass here bc one of the classes that uses `ParentContext` is in vembda